libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -34,6 +34,22 @@ var unifiedSubsystems = subsystemSet{
 	&fs.PidsGroupV2{},
 }
 
+func (m *UnifiedManager) ThawAll() error {
+	err := m.Freeze(configs.Thawed)
+	if err != nil {
+		return err
+	}
+	path, err := getSubsystemPath(m.Cgroups, "freezer")
+	if err != nil {
+		return err
+	}
+	freezer, err := unifiedSubsystems.Get("freezer")
+	if err != nil {
+		return err
+	}
+	return freezer.(*fs.FreezerGroup).RecursiveThaw(path)
```

Comment:
This needs to be a cast to `FreezerGroupV2`? And then of course `FreezerGroupV2` also needs it's new `RecursiveThaw` method.

Reply:
Yes, you can use fs cgroup method in cgroupv2, even in systemd cgroup, it'll work, but that'll be a dependency, and it'll be broken if one day we remove legacy cgroup support (though that most likely won't happen).

Reply:
I'm not sure I understand. My impression is that, if this line here is a bad cast because the underlying type will be a `FreezerGroupV2`. If we change this to a `FreezerGroupV2` cast, and then add a `RecursiveThaw` method to `FreezerGroupV2` (by analogy to the change to `FreezerGroup`), then the content of this MR will support cgroupv2 and will *not* be broken if cgroupv1 support is removed.

Reply:
That's correct.